src/runtimes/go/index.js

```diff
@@ -0,0 +1,86 @@
+const { join, extname, dirname, basename } = require('path')
+
+const cpFile = require('cp-file')
+
+const { RUNTIME_GO } = require('../../utils/consts')
+const { cachedLstat, cachedReaddir } = require('../../utils/fs')
+const { detectBinaryRuntime } = require('../detect_runtime')
+
+const { build } = require('./builder')
+
+const MAIN_FILE_NAME = 'main.go'
+
+const detectGoFunction = async ({ fsCache, path }) => {
+  const stat = await cachedLstat(fsCache, path)
+
+  if (!stat.isDirectory()) {
+    return false
+  }
+
+  const files = await cachedReaddir(fsCache, path)
+
+  return files.includes(MAIN_FILE_NAME)
```

Comment:
you could also use `go.mod` / `go.sum` as an indicator

Reply:
Would you make that a requirement? i.e. Forcing functions to have both a `main.go` _and_ a `go.mod` + `go.sum`? Could you not have a `vendor` directory instead?

Reply:
not a requirement, just a signal